Modified  build.sh to also run testasroot target

ashahi1 opened the pull request:
Fixes https://github.com/vmware/docker-volume-vsphere/issues/800

ashahi1 commented:
misc/scripts/build.sh is not running make on testasroot target.

Main job of build.shell is to run containers and then do 'make Target' within those containers.

Due to some earlier change in build.sh, we had missed to add condition to also run 'testsasroot' target while running our tests locally.

This issue was not discovered on drone because while running tests on drone, we do not call build.sh

kerneltime commented:
What was the testing done?

kerneltime commented on misc/scripts/build.sh:
```diff
@@ -122,7 +122,8 @@ then
 then
   $DOCKER run --rm -it -v $PWD/..:$dir -w $dir $pylint_container $MAKE_ESX pylint
-elif [ "$1" == "build" ] || [ "$1" == "clean-as-root" ]
+elif [ "$1" == "build" ] || [ "$1" == "clean-as-root" ] || [ "$1" == "testasroot" ]
```
We should make this the default condition, unless a special rule exists run make using the $plug_container and the $MAKE cmd. Change this to ```else``` instead of ```elseif```

ashahi1 commented:
Addressed the review comment of just having an else condition which will accept all targets.

Ran the test locally with ESX and two VMs and tests were working fine.

==========================

I could also see in the log that testasroot target is being executed  - 

../misc/scripts/build.sh testasroot
make: Entering directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'

=> Running target testasroot Wed Dec 7 18:22:29 UTC 2016

GO15VENDOREXPERIMENT=1 go test github.com/vmware/docker-volume-vsphere/vmdk_plugin/vmdkops
ok      github.com/vmware/docker-volume-vsphere/vmdk_plugin/vmdkops     1.050s
GO15VENDOREXPERIMENT=1 go test github.com/vmware/docker-volume-vsphere/vmdk_plugin/utils/config
ok      github.com/vmware/docker-volume-vsphere/vmdk_plugin/utils/config        0.005s
make: Leaving directory `/go/src/github.com/vmware/docker-volume-vsphere/vmdk_plugin'

kerneltime commented:
LGTM post CI run.
